crates/pixi_manifest/src/toml/build_system.rs

```diff
@@ -157,6 +162,24 @@ impl<'de> toml_span::Deserialize<'de> for TomlPackageBuild {
             });
         let additional_dependencies = th.optional("additional-dependencies").unwrap_or_default();
 
+        let src = th
+            .take("src")
+            .map(|(_, mut value)| {
+                // Parse the SourceSpec using its serde deserialization
+                let value_as_json = convert_toml_to_serde(&mut value)?;
```

Comment:
This seems rather ugly, cant we extend the current toml pixispec parser for this usecase?